### PR TITLE
fix non agg query

### PR DESCRIPTION
**Pull request by shz117**
- fix sql parser populating LIMIT
 - fix result flushing

### CODE REVIEW

File: query/sql_parser.go [hunk 1]
```diff
@@ -354,9 +354,11 @@ func (v *ASTBuilder) VisitQueryNoWith(ctx *antlrgen.QueryNoWithContext) interfac
 	v.SQL2AqlCtx.exprOrigin = ExprOriginOthers
 	var orderBy = v.getOrderBy(ctx)
 
-	limit, err := strconv.Atoi(v.GetTextIfPresent(ctx.GetLimit()))
-	if err != nil {
-		v.Logger.Panic("failed to parse limit")
+	limit := 0
+	if limitParsed, err := strconv.Atoi(v.GetTextIfPresent(ctx.GetLimit())); err != nil {
+		v.Logger.Warn("failed to parse limit")
```
Comment (voyager-dw) [inline query/sql_parser.go]: this should be a parse error and return

Comment (shz117) [reply query/sql_parser.go]: yeah.. I've updated the panic msg to contain more context. the panic will be recovered and throw to handler as error.

**Comment by shz117** [comment]
looks like the coverage diff is not correct, will merge..

File: query/sql_parser.go [hunk 2]
```diff
@@ -354,12 +354,22 @@ func (v *ASTBuilder) VisitQueryNoWith(ctx *antlrgen.QueryNoWithContext) interfac
 	v.SQL2AqlCtx.exprOrigin = ExprOriginOthers
 	var orderBy = v.getOrderBy(ctx)
 
+	limit := 0
+	limitNode := ctx.GetLimit()
+	if limitParsed, err := strconv.Atoi(v.GetTextIfPresent(limitNode)); limitNode != nil && err != nil {
+		limitLoc := v.getLocation(limitNode)
+		v.Logger.Panicf("failed to parse limit %s at (line:%d, col:%d)", limitNode.GetText(), limitLoc.Line, limitLoc.CharPosition)
```
Comment (voyager-dw) [inline query/sql_parser.go]: oh no error to return here? panic will be too bad, or recover outside?

**Pull request merged**